macaroons/service.go

```diff
@@ -208,6 +215,23 @@ func (svc *Service) CheckMacAuth(ctx context.Context, macBytes []byte,
 		return err
 	}
 
+	// Ensure that the macaroon is using the exact same version as we
+	// expect. In the future, we can relax this check to phase in new
+	// versions.
+	if mac.Version() != macaroon.V2 {
+		return fmt.Errorf("%w: %v", ErrUnknownVersion,
+			mac.Version())
+	}
+
+	// Run a similar version check on the ID used for the macaroon as well.
+	const minIDLength = 1
+	if len(mac.Id()) < minIDLength {
+		return ErrInvalidID
+	}
+	if mac.Id()[0] != byte(bakery.LatestVersion) {
```

Comment:
Should we explicitly check the version is above `bakery.Version3` instead?  Using `bakery.LatestVersion` implicitly puts the version managing at the `bakery` package side, which may get updated without our notice. There's also compatibility concern.

Reply:
The packages are pinned, but I still think it's a good idea to explicitly set it to `bakery.Version3`